fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2909,7 +2918,29 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
         LOG.info("Successfully create materialized view[{};{}]", mvName, mvId);
 
         // NOTE: The materialized view  has been added to the database, and the following procedure cannot throw exception.
-        // todo add async task after task framework is completed
+        if (createMvSuccess) {
+            if (materializedView.getRefreshScheme().getType() == MaterializedView.RefreshType.ASYNC) {
+                // create task
+                Task task = new Task();
+                long taskId = GlobalStateMgr.getCurrentState().getNextId();
+                task.setId(taskId);
+                task.setName("mv-" + materializedView.getId());
+                task.setCreateTime(System.currentTimeMillis());
+                task.setDbName(getDb(materializedView.getDbId()).getFullName());
+                Map<String, String> taskProperties = Maps.newHashMap();
+                taskProperties.put("mvId", String.valueOf(materializedView.getId()));
+                task.setProperties(taskProperties);
+                task.setDefinition(materializedView.getViewDefineSql());
+                task.setExpireTime(0L);
+                TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
+                taskManager.createTask(task, true);
+                // run task
+                TaskRun taskRun = TaskRunBuilder.newBuilder(task).build();
+                taskRun.setProcessor(new MvTaskRunProcessor());
+                TaskRunManager taskRunManager = taskManager.getTaskRunManager();
```

Comment:
You should executeTask here instead of set MvTaskRunProcessor.
This is because other unified entries for triggering scheduling are executeTask, which is controlled by the Task framework. You may need to modify the Task framework to add a source field to mark that if it is a CTAS, use sqlTaskRunProcesser, if it is a materialized view, create a MvTaskRunProcessor.